pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -246,6 +264,49 @@ func (s *SnapshotScheduleController) updateVolumeSnapshotStatus(snapshotSchedule
 	return nil
 }
 
+// cleanupErroredSnapshots cleans any volumesnapshot CRs that are in Error state. It doesn't remove them
+// from the volumesnapshotschedule items list though.
+func (s *SnapshotScheduleController) cleanupErroredSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) (err error) {
+	// Filter out the errored out snapshots that are older than the cleanup cutoff period.
+	for policy, policyVolumeSnapshot := range snapshotSchedule.Status.Items {
+		for idx, snapshot := range policyVolumeSnapshot {
+			if snapshot.Status == snapv1.VolumeSnapshotConditionError && time.Since(snapshot.CreationTimestamp.Time) > errorSnapshotCleanupCutoffTime {
+				// Fetch the latest status as well in order to mitigate any late volumesnapshot updates.
+				snapshotStatus, err := getVolumeSnapshotStatus(snapshot.Name, snapshotSchedule.Namespace)
+				if err != nil {
+					log.VolumeSnapshotScheduleLog(snapshotSchedule).Warnf("Error getting status of snapshot %s: %s", snapshot.Name, err.Error())
+					continue
+				}
+
+				if snapshotStatus == snapv1.VolumeSnapshotConditionError {
+					log.VolumeSnapshotScheduleLog(snapshotSchedule).Infof("Going to delete the errored out snapshot: %v, age: %v", snapshot.Name, time.Since(snapshot.CreationTimestamp.Time))
+					err := k8sextops.Instance().DeleteSnapshot(snapshot.Name, snapshotSchedule.Namespace)
+					if err != nil && !errors.IsNotFound(err) {
+						log.VolumeSnapshotScheduleLog(snapshotSchedule).Warnf("Error deleting errored out snapshot %v: %v", snapshot.Name, err)
+						continue
+					}
+
+					// Update the volumesnapshotschedule object with the deletion message and setting `deleted` to `true`.
+					snapshotSchedule.Status.Items[policy][idx].Deleted = true
+					snapshotSchedule.Status.Items[policy][idx].Message = errorSnapshotDeletionMessage
```

Comment:
let's copy the error message from the actual snapshot instead of putting this hardcoded message.
Also let's log a line before deleting the snapshot stating the snapshot status and the error message, in case the volumesnapshot deletion happens but then the update to volumesnapshotschedule could not happen.

Reply:
Done